diff --git a/kernel/supercall/dispatch.c b/kernel/supercall/dispatch.c
index 70910c51..41272ba1 100644
--- a/kernel/supercall/dispatch.c
+++ b/kernel/supercall/dispatch.c
@@ -383,7 +383,7 @@ static int do_set_app_profile(void __user *arg)
     ret = ksu_set_app_profile(&cmd.profile);
     if (!ret) {
         ksu_persistent_allow_list();
-#ifdef CONFIG_KSU_TRACEPOINT_HOOK
+#if defined(CONFIG_KSU_TRACEPOINT_HOOK) && defined(CONFIG_KSU_SUSFS)
         ksu_mark_running_process();
 #endif
     }
@@ -994,6 +994,12 @@ int ksu_handle_susfs_cmd(unsigned int cmd, void __user **arg)
         return 0;
     }
 #endif //#ifdef CONFIG_KSU_SUSFS_SUS_KSTAT
+#ifdef CONFIG_KSU_SUSFS_SUS_KSTAT_REDIRECT
+    case CMD_SUSFS_ADD_SUS_KSTAT_REDIRECT: {
+        susfs_add_sus_kstat_redirect(arg);
+        return 0;
+    }
+#endif
 #ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
     case CMD_SUSFS_SET_UNAME: {
         susfs_set_uname(arg);
